### PR TITLE
Next-36695 Trasform Scaffold generator into their own separated commands

### PR DESCRIPTION
**Pull request by raffaelecarelle**
### 1. Why is this change necessary?
This change can speed on development process, creating scaffold during whole development process avoiding copy paste errors

### 2. What does this change do, exactly?
Trasnform all scaffold generator into separated command

### 3. Describe each step to reproduce the issue or behaviour.
---

### 4. Please link to the relevant issues (if any).
---

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [x] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Core/Content/Maker/Command/MakerCommand.php [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+                return self::FAILURE;
+            }
+
+            $plugin = $this->pluginService->getPluginByName($pluginName, Context::createCLIContext());
+
+            $directory = $plugin->getPath();
+
+            if ($directory === null) {
+                $io->error('Plugin bnase path is null');
```
Comment (shyim) [inline src/Core/Content/Maker/Command/MakerCommand.php]: typo

Comment (raffaelecarelle) [reply src/Core/Content/Maker/Command/MakerCommand.php]: Hi @shyim, fixed

File: src/Core/Content/Maker/Command/MakerCommand.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            $input->setArgument($argument->getName(), $value);
+        }
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        try {
+            $pluginName = $input->getArgument('plugin-name');
```
Comment (keulinho) [inline src/Core/Content/Maker/Command/MakerCommand.php:74]: Can you come up with a test case that will cover your new commands? So that we can ensure they will continue working, even when we might touch & refactor the scaffolding system?

File: src/Core/Content/Maker/DependencyInjection/CompilerPass/CreateGeneretorScafoldingCommandPass.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        $value = str_replace('_', '-', self::asSnake($value));
+
+        return str_replace('-generator', '', $value);
+    }
+
+    /**
+     * (e.g. `BlogPostType` -> `blog_post_type`).
+     */
+    private static function asSnake(string $value): string
```
Comment (keulinho) [inline src/Core/Content/Maker/DependencyInjection/CompilerPass/CreateGeneretorScafoldingCommandPass.php:59]: Symfony already has a `CamelCaseToSnakeCaseNameConverter` can we use that instead?

File: src/Core/Content/Maker/DependencyInjection/CompilerPass/CreateGeneretorScafoldingCommandPass.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+use Shopware\Core\Framework\Plugin\Command\Scaffolding\Generator\ScaffoldingGenerator;
+use Symfony\Component\DependencyInjection\ChildDefinition;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+#[Package('content')]
+class CreateGeneretorScafoldingCommandPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
```
Comment (keulinho) [inline src/Core/Content/Maker/DependencyInjection/CompilerPass/CreateGeneretorScafoldingCommandPass.php:17]: can you also add a unit test for the compiler pass? You should be able to create a dummy container from scratch that is properly set up for the cases you want to test and then run this process method and check that the commands are registered as expected in the container

**Review by keulinho — CHANGES_REQUESTED** [review]
Hey thanks for your contribution! :tada: 
Really nice improvement, didn't though about that.

We just have a few remarks ;)

File: src/Core/Content/Maker/DependencyInjection/CompilerPass/CreateGeneretorScafoldingCommandPass.php (hunk 3 above)
Comment (develop-algoritma) [reply src/Core/Content/Maker/DependencyInjection/CompilerPass/CreateGeneretorScafoldingCommandPass.php:57]: Good idea, thank you

Comment (raffaelecarelle) [reply src/Core/Content/Maker/DependencyInjection/CompilerPass/CreateGeneretorScafoldingCommandPass.php:59]: Good idea, thank you!

File: src/Core/Content/Maker/DependencyInjection/CompilerPass/CreateGeneretorScafoldingCommandPass.php (hunk 4 above)
Comment (raffaelecarelle) [reply src/Core/Content/Maker/DependencyInjection/CompilerPass/CreateGeneretorScafoldingCommandPass.php:17]: @keulinho, done

File: src/Core/Content/Maker/DependencyInjection/CompilerPass/CreateGeneretorScafoldingCommandPass.php (hunk 3 above)
Comment (raffaelecarelle) [reply src/Core/Content/Maker/DependencyInjection/CompilerPass/CreateGeneretorScafoldingCommandPass.php:59]: @keulinho, ok done

File: src/Core/Content/Maker/Command/MakerCommand.php (hunk 2 above)
Comment (raffaelecarelle) [reply src/Core/Content/Maker/Command/MakerCommand.php:74]: @keulinho, ok done

File: composer.json [hunk 5]
```diff
@@ -407,7 +407,7 @@
             "symfony local:server:start --dir=tests/e2e/update-api-mock --port=8060 -d"
         ],
         "ecs": "ecs check src tests",
-        "ecs-fix": "ecs check --fix src tests",
+        "ecs-fix": "ecs check --fix",
```
Comment (mitelg) [inline composer.json]: why did you remove that? :thinking:

Comment (raffaelecarelle) [reply composer.json]: oh sorry, my bad. I have removed this changes

**Review by keulinho — APPROVED** [review]

**Comment by keulinho** [comment]
Thanks again for your contribution :tada: Your PR got merged, i did some additional polishing, e.g. adding some tests, moving the new stuff to the framework domain and adding a changelog.

The only meaningful change is that we changed the command prefix from just `make:` to `make:plugin:` as that is more specific, :v:

**Comment by Isengo1989** [comment]
Small note, even if this is quite an old PR. Give us @shopware/pp-developer-experience a quick nodge - great commands for developers that were not documented yet :writing_hand:

**Comment by raffaelecarelle** [comment]
Thanks @Isengo1989, but what is it? I can't find it anywhere..

**Comment by Isengo1989** [comment]
> Thanks @Isengo1989, but what is it? I can't find it anywhere..

Do you mean the docs repo or something else? No worries on your part, you did a great job already :) 

I added a PR here -> https://github.com/shopware/docs/pull/2031/files#diff-db4955cc7e092e563353cee3841e10b255363ecbecb59c3d62204484b281bc23R227
